### PR TITLE
Module6

### PR DESCRIPTION
**Pull request by SerhiiSmilianets**

### CODE REVIEW

File: src/components/DefaultHeader.jsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+                <DialogOpenButton dialogTitle={"Add Movie"} buttonText={"+ ADD MOVIE"}  onDialogFormSubmit={() => {}}/>
+            </div>
+            <div className="searchbar__container">
+                <SearchBar/>
+            </div>
+        </div>
+    )
+}
+
+export default DefaultHeader
```
Comment (MykhailoSheliahov) [inline src/components/DefaultHeader.jsx:17]: Add empty line

File: src/components/DefaultHeader.jsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import SearchBar from './SearchBar';
+import DialogOpenButton from './DialogOpenButton';
+
+const DefaultHeader = () => {
+    return (
+        <div className="default-header__container">
+            <div className="add-movie__container">
+                <DialogOpenButton dialogTitle={"Add Movie"} buttonText={"+ ADD MOVIE"}  onDialogFormSubmit={() => {}}/>
```
Comment (MykhailoSheliahov) [inline src/components/DefaultHeader.jsx:8]:
```suggestion
                <DialogOpenButton dialogTitle={"Add Movie"} buttonText={"+ ADD MOVIE"}  handleSubmit={() => {}}/>
```

File: src/components/GenreList.jsx [hunk 3]
```diff
@@ -1,20 +1,38 @@
...
     return (
         <li>
-            <button className={"genre-item " + (isSelected ? "selected" : "")} onClick={() => onGenreSelection(genre)} >{genre} {isSelected}</button>
+            <button className={"genre-item " + (selectedGenre === genre ? "selected" : "")} onClick={() => onGenreSelection(genre)} >{genre}</button>
         </li>
     )
 }
 
-const GenreList = ({ genresList, selectedGenre, handleSelectGenre }) => {
+const GenreList = ({ genresList }) => {
```
Comment (MykhailoSheliahov) [inline src/components/GenreList.jsx:13]: add type for genreList

File: src/components/GenreList.jsx [hunk 4]
```diff
@@ -1,20 +1,38 @@
...
+    const [searchParams, setSearchParams] = useSearchParams();
+
+    const handleSelectGenre = (genre) => {
```
Comment (MykhailoSheliahov) [inline src/components/GenreList.jsx:16]:
```suggestion
    const handleSelectGenre = (genre:string) => {
```

File: src/components/MovieDetails.jsx [hunk 5]
```diff
@@ -26,19 +30,13 @@ const MovieDetails = ({poster_path, title, vote_average, genres, release_date, r
...
-    vote_average: PropTypes.number,
-    genres: PropTypes.arrayOf(PropTypes.string),
-    release_date: PropTypes.string,
-    runtime: PropTypes.number,
-    overview: PropTypes.string,
-}
-
 export default MovieDetails;
+
+export async function loader({ params}) {
```
Comment (MykhailoSheliahov) [inline src/components/MovieDetails.jsx:40]: add type for params

File: src/components/MovieTile.jsx [hunk 6]
```diff
@@ -20,9 +21,10 @@ const useOutsideClick = (menuRef, setMenuTileContextOpen) => {
     }, [menuRef, setMenuTileContextOpen]);
 }
 
-const MovieTile = ({movieData, onClick}) => {
+const MovieTile = ({movieData}) => {
```
Comment (MykhailoSheliahov) [inline src/components/MovieTile.jsx:24]: add type for  movieData

File: src/styles/MovieListPage.scss [hunk 7]
```diff
@@ -58,6 +58,7 @@ nav {
         flex-direction: row;
         justify-content: space-between;
         align-items: baseline;
+        // padding-bottom: 8px;
```
Comment (MykhailoSheliahov) [inline src/styles/MovieListPage.scss]: ✂

File: src/styles/MovieTile.scss [hunk 8]
```diff
@@ -6,14 +6,21 @@
     height: 100%;
     color: $text-color;
     cursor: pointer;
+    text-decoration: none;
 
     &__container {
+        position: relative;
+        z-index: 0;
```
Comment (MykhailoSheliahov) [inline src/styles/MovieTile.scss]: move it back

File: src/utils/movieUtils.js [hunk 9]
```diff
@@ -9,7 +9,15 @@ const getReleaseYear = (release_date) => {
     return (new Date(release_date).getFullYear())
 }
 
+const fetchMovies = async (url) => {
+    const response = await fetch(url);
+    const resData = await response.json();
+
+    return resData
+}
```
Comment (MykhailoSheliahov) [inline src/utils/movieUtils.js]:
```suggestion
const fetchMovies = async (url) => {
    const response = await fetch(url);
    const resData = await response.json();

    return resData.data
}
```

**Review by MykhailoSheliahov — CHANGES_REQUESTED** [review]
Please rename all .jsx, js files to .tsx, .ts, and add types if needed

File: src/components/GenreList.jsx [hunk 10]
```diff
@@ -1,20 +1,38 @@
...
+            if  (key !== "filter") {
+                queryObj[key] = searchParams.get(key);
+            }
+        }
+
+        if (genre === "All") {
+            setSearchParams({...queryObj})
+        } else {
+            setSearchParams({...queryObj, filter: genre})
+        }
```
Comment (MykhailoSheliahov) [inline src/components/GenreList.jsx:28]:
```suggestion
        if (genre === "All") {
            newSearchParams.delete('filter');
        } else {
            newSearchParams.set('filter', genre);
        }
```

File: src/utils/movieUtils.js (hunk 9 above)
Comment (SerhiiSmilianets) [reply src/utils/movieUtils.js]: @MykhailoSheliahov it's universal function for the get movie by id and get movie list. When get movie by id it returns already object for movie, when we get list - it returns object with additional data and also array of movies.

File: src/components/GenreList.jsx (hunk 10 above)
Comment (SerhiiSmilianets) [reply src/components/GenreList.jsx:28]: @MykhailoSheliahov in this case other query parameters will disappear, I tried solution you provided

**Comment by SerhiiSmilianets** [comment]
> Please rename all .jsx, js files to .tsx, .ts, and add types if needed

Module 10 has task to move all code to typescript

File: src/components/DefaultHeader.jsx (hunk 1 above)
Comment (SerhiiSmilianets) [reply src/components/DefaultHeader.jsx:17]: default header is removed in module 7 https://github.com/SerhiiSmilianets/reactjs-global-mentoring-program/pull/7/files

File: src/components/DefaultHeader.jsx (hunk 2 above)
Comment (SerhiiSmilianets) [reply src/components/DefaultHeader.jsx:8]: default header is removed in module 7 https://github.com/SerhiiSmilianets/reactjs-global-mentoring-program/pull/7/files

File: src/components/GenreList.jsx (hunk 3 above)
Comment (SerhiiSmilianets) [reply src/components/GenreList.jsx:13]: Module 10 has task to move all code to typescript

File: src/components/GenreList.jsx (hunk 4 above)
Comment (SerhiiSmilianets) [reply src/components/GenreList.jsx:16]: Module 10 has task to move all code to typescript

File: src/components/MovieDetails.jsx (hunk 5 above)
Comment (SerhiiSmilianets) [reply src/components/MovieDetails.jsx:40]: Module 10 has task to move all code to typescript

File: src/components/MovieTile.jsx (hunk 6 above)
Comment (SerhiiSmilianets) [reply src/components/MovieTile.jsx:24]: Module 10 has task to move all code to typescript

File: src/components/DefaultHeader.jsx (hunk 1 above)
Comment (MykhailoSheliahov) [reply src/components/DefaultHeader.jsx:17]: ok

File: src/utils/movieUtils.js (hunk 9 above)
Comment (MykhailoSheliahov) [reply src/utils/movieUtils.js]: @SerhiiSmilianets should we create 2 function  fetchMovie and fetchMovies to separate logic?

**Review by MykhailoSheliahov — APPROVED** [review]

**Pull request merged**